packages/headless/src/modules/brands/services/index.ts

```diff
@@ -0,0 +1,64 @@
+
+import { HttpStatus, Injectable } from "@nestjs/common";
+import { Response } from "express";
+import * as crypto from 'crypto';
+import { ErrorMessage, ErrorMessageDeleteBrand, ErrorMessageGetBrandById, ErrorMessageUpdate } from "models";
+
+import { Brand } from 'src/entity/brand';
+import { BrandRepository } from './../repositories/index';
+import { CreateBrandRequestDto, CreateBrandResponseDto } from 'src/modules/brands/dto/createBrandDto';
+import { GetAllBrandsResponseDto } from "src/modules/brands/dto/getAllBrandsDto";
+import { UpdateBrandRequestdto, UpdateBrandResponseDto } from "src/modules/brands/dto/updateBrandDto";
+import { BrandDto } from 'src/modules/brands/dto/brandDto';
+import { GetBrandByIdResponseDto } from "src/modules/brands/dto/getBrandByIdDto";
+import { DeleteBrandResponseDto } from "src/modules/brands/dto/deleteBrandDto";
+
+@Injectable()
+
+export class BrandService{
+    constructor( private brandRepo: BrandRepository ){}
+
+    async createBrand( brand: CreateBrandRequestDto ): Promise<CreateBrandResponseDto>{
+            const doesBrandExist = await this.brandRepo.getBrandByName(brand.info.name);
+            if(doesBrandExist) return { error: ErrorMessage.BRAND_ALREADY_EXISTS, errors: null, code: HttpStatus.BAD_REQUEST};
```

Comment:
you can write code here without an if-else block.

Reply:
how? can you please elaborate